apl-integration-tests/src/test/java/com/apollocurrrency/aplwallet/inttest/model/TestBaseOld.java

```diff
@@ -842,7 +884,175 @@ public void createPhasingVote(long phasingFinishHeight, Parameters votingModel,
     }
 
 
+    @Step("Get Dex History with param: Type: {2}")
+    public CreateTransactionResponse issueCurrency(Wallet wallet,int type, String name, String description, String code, int initialSupply,int maxSupply, int decimals){
+        int currentHeight = getBlock().getHeight();
+        int finishHeight = currentHeight + 200;
+        int issuanceHeight = currentHeight + 4;
+        addParameters(RequestType.requestType,RequestType.issueCurrency);
+        addParameters(Parameters.name, name);
+        addParameters(Parameters.code ,code );
+        addParameters(Parameters.description, description);
+        addParameters(Parameters.type, type);
+        addParameters(Parameters.initialSupply, initialSupply);
+        addParameters(Parameters.decimals, decimals);
+        addParameters(Parameters.feeATM, 10000000000L);
+        addParameters(Parameters.deadline, "1440");
+        addParameters(Parameters.wallet, wallet);
+
+        switch (type){
```

Comment:
Maybe better to use bitwise operations to determine correct currency types?
EXCHANGEABLE - 1
CONTROLLABLE  - 2
RESERVABLE  - 4
CLAIMABLE  - 8
MINTABLE  - 16
NON_SHUFFLEABLE - 32

Thus if currency is claimable+reservable+exchangeable, then binary representation of type = 1101 or 1 + 4 + 8 = 13
You can check packed int type for presence of any listed above type by using binary mask: packed_type & (currency_type) == currency_type

Reply:
Done